src/dagmc.cpp

```diff
@@ -69,35 +159,80 @@ void load_dagmc_geometry()
       model::universes[it->second]->cells_.push_back(i);
     }
 
+    // check for temperature assignment
+    std::string temp_value;
+    if (model::DAG->has_prop(vol_handle, "temp")) {
+      rval = model::DAG->prop_value(vol_handle, "temp", temp_value);
+      MB_CHK_ERR_CONT(rval);
+      double temp = std::stod(temp_value);
+      c->sqrtkT_.push_back(std::sqrt(K_BOLTZMANN * temp));
+    } else {
+      c->sqrtkT_.push_back(std::sqrt(K_BOLTZMANN * settings::temperature_default));
+    }
+
+    // MATERIALS
+
     if (model::DAG->is_implicit_complement(vol_handle)) {
-      // assuming implicit complement is void for now
-      c->material_.push_back(MATERIAL_VOID);
+      if (model::DAG->has_prop(vol_handle, "mat")) {
+        // if the implicit complement has been assigned a material, use it
+        std::string comp_mat = DMD.volume_material_property_data_eh[vol_handle];
+        // Note: material numbers are set by UWUW
+        int mat_number = uwuw.material_library[comp_mat].metadata["mat_number"].asInt();
+        c->material_.push_back(mat_number);
+      } else {
+        // if no material is found, the implicit complement is void
+        c->material_.push_back(MATERIAL_VOID);
+      }
       continue;
     }
 
-    if (model::DAG->has_prop(vol_handle, "mat")){
-      std::string mat_value;
+    // determine volume material assignment
+    std::string mat_value;
+    if (model::DAG->has_prop(vol_handle, "mat")) {
       rval = model::DAG->prop_value(vol_handle, "mat", mat_value);
       MB_CHK_ERR_CONT(rval);
-      to_lower(mat_value);
-
-      if (mat_value == "void" || mat_value == "vacuum") {
-        c->material_.push_back(MATERIAL_VOID);
-      } else {
-        c->material_.push_back(std::stoi(mat_value));
-      }
     } else {
       std::stringstream err_msg;
       err_msg << "Volume " << c->id_ << " has no material assignment.";
       fatal_error(err_msg.str());
     }
+
+    std::string cmp_str = mat_value;
+    to_lower(cmp_str);
+    // material void checks
+    if (cmp_str.find("void") != std::string::npos   ||
+        cmp_str.find("vacuum") != std::string::npos ||
+        cmp_str.find("graveyard") != std::string::npos) {
+      c->material_.push_back(MATERIAL_VOID);
+    } else {
+      if (using_uwuw) {
+        // lookup material in uwuw if the were present
+        std::string uwuw_mat = DMD.volume_material_property_data_eh[vol_handle];
+        if (uwuw.material_library.count(uwuw_mat) != 0) {
+          // Note: material numbers are set by UWUW
+          int mat_number = uwuw.material_library[uwuw_mat].metadata["mat_number"].asInt();
+          c->material_.push_back(mat_number);
+        } else {
+          std::stringstream err_msg;
+          err_msg << "Material with value " << mat_value << " not found ";
+          err_msg << "in the UWUW material library";
+          fatal_error(err_msg.str());
```

Comment:
We actually have an overload for `stringstream`, so you can just call `fatal_error(err_msg);`